Perl syntax highlighting inside double quotes

### Summary

Perl syntax highlighting breaks inside double quotes.

### Reproduction Steps

<!-- Ideally provide a key sequence and/or asciinema.org recording. --> 

I tried this:

1. `hx test.pl`

I expected this to happen:

Syntax hightliting should not break inside "".

Instead, this happened:

Syntax hightliting breaks  inside "".

![Screenshot from 2023-12-12 09-52-09](https://github.com/helix-editor/helix/assets/3610557/caca6edc-8b7f-4009-87b1-888bffa46b85)

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-08-11T09:55:54.321 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-08-11T09:56:09.712 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-08-11T09:56:19.698 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-08-11T09:57:10.443 helix_view::editor [ERROR] Failed to initialize the LSP for `source.js` { cannot find binary path }
2023-08-11T09:57:25.142 helix_view::theme [WARN] Theme: malformed hexcode: light-black
2023-08-11T09:57:25.142 helix_view::theme [WARN] Theme: malformed hexcode: light-black
2023-08-11T09:57:25.142 helix_view::theme [WARN] Theme: malformed hexcode: purple
2023-08-11T09:57:25.653 helix_view::theme [WARN] Theme: malformed hexcode: light-black
2023-08-11T09:57:25.653 helix_view::theme [WARN] Theme: malformed hexcode: light-black
2023-08-11T09:57:25.653 helix_view::theme [WARN] Theme: malformed hexcode: purple
2023-08-11T09:57:54.021 helix_view::theme [WARN] Theme: malformed hexcode: light-black
2023-08-11T09:57:54.022 helix_view::theme [WARN] Theme: malformed hexcode: light-black
2023-08-11T09:57:54.022 helix_view::theme [WARN] Theme: malformed hexcode: purple
2023-08-11T09:58:10.908 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-08-11T09:58:15.218 helix_view::editor [ERROR] editor error: Register [@] empty
2023-08-11T09:58:19.035 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
2023-08-11T09:58:38.289 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { cannot find binary path }
2023-08-11T09:59:12.861 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { cannot find binary path }
2023-08-11T10:00:19.157 helix_view::editor [ERROR] Failed to initialize the LSP for `source.python` { cannot find binary path }
2023-08-11T10:00:39.445 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: ["Documents/uart-loop.py"]
2023-08-11T10:00:43.429 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: ["Documents/uart-loop.py"]
2023-08-11T10:06:45.692 helix_lsp::transport [ERROR] err <- "standard-module-name-resolver: collection not found\n"
2023-08-11T10:06:45.694 helix_lsp::transport [ERROR] err <- "  for module path: (lib \"racket-langserver\")\n"
2023-08-11T10:06:45.694 helix_lsp::transport [ERROR] err <- "  collection: \"racket-langserver\"\n"
2023-08-11T10:06:45.694 helix_lsp::transport [ERROR] err <- "  in collection directories:\n"
2023-08-11T10:06:45.694 helix_lsp::transport [ERROR] err <- "   ... [166 additional linked and package directories]\n"
2023-08-11T10:06:45.695 helix_lsp [ERROR] failed to initialize language server: server closed the stream
2023-08-11T10:06:45.695 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-08-11T10:06:50.351 helix_lsp::transport [ERROR] err <- "standard-module-name-resolver: collection not found\n"
2023-08-11T10:06:50.352 helix_lsp::transport [ERROR] err <- "  for module path: (lib \"racket-langserver\")\n"
2023-08-11T10:06:50.352 helix_lsp::transport [ERROR] err <- "  collection: \"racket-langserver\"\n"
2023-08-11T10:06:50.352 helix_lsp::transport [ERROR] err <- "  in collection directories:\n"
2023-08-11T10:06:50.352 helix_lsp::transport [ERROR] err <- "   ... [166 additional linked and package directories]\n"
2023-08-11T10:06:50.353 helix_lsp [ERROR] failed to initialize language server: server closed the stream
2023-08-11T10:06:50.353 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-08-11T10:06:54.999 helix_lsp::transport [ERROR] err <- "standard-module-name-resolver: collection not found\n"
2023-08-11T10:06:55.000 helix_lsp::transport [ERROR] err <- "  for module path: (lib \"racket-langserver\")\n"
2023-08-11T10:06:55.000 helix_lsp::transport [ERROR] err <- "  collection: \"racket-langserver\"\n"
2023-08-11T10:06:55.000 helix_lsp::transport [ERROR] err <- "  in collection directories:\n"
2023-08-11T10:06:55.000 helix_lsp::transport [ERROR] err <- "   ... [166 additional linked and package directories]\n"
2023-08-11T10:06:55.001 helix_lsp [ERROR] failed to initialize language server: server closed the stream
2023-08-11T10:06:55.001 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-08-11T10:07:17.687 helix_lsp::transport [ERROR] err <- "standard-module-name-resolver: collection not found\n"
2023-08-11T10:07:17.689 helix_lsp::transport [ERROR] err <- "  for module path: (lib \"racket-langserver\")\n"
2023-08-11T10:07:17.689 helix_lsp::transport [ERROR] err <- "  collection: \"racket-langserver\"\n"
2023-08-11T10:07:17.689 helix_lsp::transport [ERROR] err <- "  in collection directories:\n"
2023-08-11T10:07:17.689 helix_lsp::transport [ERROR] err <- "   ... [166 additional linked and package directories]\n"
2023-08-11T10:07:17.689 helix_lsp [ERROR] failed to initialize language server: server closed the stream
2023-08-11T10:07:17.689 helix_lsp::transport [ERROR] err: <- StreamClosed
2023-08-11T10:07:39.722 helix_view::editor [ERROR] Failed to initialize the LSP for `source.js` { cannot find binary path }
2023-08-11T10:09:00.539 helix_view::editor [ERROR] Failed to initialize the LSP for `source.js` { cannot find binary path }
2023-12-12T09:47:20.451 helix_view::theme [WARN] Theme: malformed ANSI: bg3
2023-12-12T09:47:58.098 helix_view::theme [WARN] Theme: malformed ANSI: bg3
2023-12-12T09:47:58.265 helix_view::editor [ERROR] Failed to initialize the language servers for `source.perl` - `perlnavigator` { cannot find binary path }
2023-12-12T09:48:02.611 helix_view::theme [WARN] Theme: malformed ANSI: bg3
2023-12-12T09:48:02.777 helix_view::editor [ERROR] Failed to initialize the language servers for `source.perl` - `perlnavigator` { cannot find binary path }
```

</details>


### Platform

Linux 6.2.0-37-generic #38~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Nov  2 18:01:13 UTC 2 x86_64 x86_64 GNU/Linux

### Terminal Emulator

GNOME Terminal 3.44.0 using VTE 0.68.0 +BIDI +GNUTLS +ICU +SYSTEMD

### Installation Method

Releases page appimge

### Helix Version

helix 23.10 (f6021dd0)

--- a/languages.toml
+++ b/languages.toml
@@ -1469,7 +1469,7 @@ indent = { tab-width = 2, unit = "  " }
 
 [[grammar]]
 name = "perl"
-source = { git = "https://github.com/tree-sitter-perl/tree-sitter-perl", rev = "e99bb5283805db4cb86c964722d709df21b0ac16" }
+source = { git = "https://github.com/tree-sitter-perl/tree-sitter-perl", rev = "81c57e274c541c30d6c204c8dbceb90ec647ed8d" }
 
 [[language]]
 name = "pod"
@@ -1479,7 +1479,7 @@ file-types = ["pod"]
 
 [[grammar]]
 name = "pod"
-source = { git = "https://github.com/tree-sitter-perl/tree-sitter-pod", rev = "39da859947b94abdee43e431368e1ae975c0a424" }
+source = { git = "https://github.com/tree-sitter-perl/tree-sitter-pod", rev = "0bf8387987c21bf2f8ed41d2575a8f22b139687f" }
 
 [[language]]
 name = "racket"

--- a/languages.toml
+++ b/languages.toml
@@ -1469,7 +1469,7 @@ indent = { tab-width = 2, unit = "  " }
 
 [[grammar]]
 name = "perl"
-source = { git = "https://github.com/tree-sitter-perl/tree-sitter-perl", rev = "81c57e274c541c30d6c204c8dbceb90ec647ed8d" }
+source = { git = "https://github.com/tree-sitter-perl/tree-sitter-perl", rev = "72a08a496a23212f23802490ef6f4700d68cfd0e" }
 
 [[language]]
 name = "pod"

--- a/runtime/queries/perl/highlights.scm
+++ b/runtime/queries/perl/highlights.scm
@@ -1,9 +1,9 @@
 [
-  "use" "no" "require" "package"
+  "use" "no" "require" "package" "class" "role"
 ] @keyword.control.import
 
 [
-  "sub"
+  "sub" "method" "async" "extended"
 ] @keyword.function
 
 [
@@ -17,7 +17,7 @@
 ] @keyword.control.repeat
 
 [
-  "my" "our" "local"
+  "my" "our" "local" "state"
 ] @keyword.storage.modifier
 
 [
@@ -29,9 +29,10 @@
 ] @constant.builtin
 
 (phaser_statement phase: _ @keyword.directive)
+(class_phaser_statement phase: _ @keyword.directive)
 
 [
-  "or" "and"
+  "or" "xor" "and"
   "eq" "ne" "cmp" "lt" "le" "ge" "gt"
   "isa"
 ] @keyword.operator
@@ -55,7 +56,7 @@
 
 [(quoted_regexp) (match_regexp)] @string.regexp
 
-(autoquoted_bareword _?) @string.special
+(autoquoted_bareword) @string.special
 
 [(scalar) (arraylen)] @variable
 (scalar_deref_expression ["->" "$" "*"] @variable)

